### PR TITLE
Port SDQL2 from tg

### PR DESCRIPTION
**Pull request by Tayyyyyyy**
## What does this PR do
Ports tg's version of SDQL2, since our version hasn't been updated in a while. This brings support for nifty things like documentation and MAP. I've taken their documentation and rewritten it as [`README.md`](https://github.com/ParadiseSS13/Paradise/blob/b6c080e54e84ce481295223a63e8c9dc3a4ac4d1/code/modules/admin/verbs/SDQL2/README.md), which will have all the details of the system.

Oh it also adds a bunch of wrapper functions to allow admins to call inbuilt functions they normally can't.

When SDQL queries are running, their statuses can be viewed in the stat panel as well as canceled:
image

### Differences from tg
The few things I had to adjust for the port:
- UID instead of REF
- tg restricts sdql2 queries from being executed via admin proc call, but we don't have this system
- ~~We don't have some SS subsystems~~ Subsystems are dynamic now

Everything else is directly copied from tg

### Authorship
The authors who contributed to this code can be viewed [here](https://github.com/tgstation/tgstation/blob/8eead98e847cc894333153808d5ee830cd8eb551/code/modules/admin/verbs/SDQL2/SDQL_2.dm).

### CODE REVIEW

**Comment by KasparoVy** [comment]
I know we got a couple SDQL wizards around here that'll appreciate this.

File: code/modules/admin/verbs/SDQL2/useful_procs.dm [hunk 1]
```diff
@@ -27,3 +27,219 @@ CALL global.json_to_object_arbitrary_vars("{'type':'/obj/item/crowbar', 'color':
...
+/proc/_abs(A)
+	return abs(A)
+
+/proc/_animate(atom/A, set_vars, time = 10, loop = 1, easing = LINEAR_EASING, flags = null)
+	var/mutable_appearance/MA = new()
+	for(var/v in set_vars)
+		MA.vars[v] = set_vars[v]
+	animate(A, appearance = MA, time, loop, easing, flags)
+
+/proc/_acrccos(A)
```
Comment (Citinited) [inline code/modules/admin/verbs/SDQL2/useful_procs.dm]: Is this a typo?

Comment (Tayyyyyyy) [reply code/modules/admin/verbs/SDQL2/useful_procs.dm]: I don't think so... It's arc cosine, so the two cs are intentional. It does look weird tho.

Comment (Lunaimon) [reply code/modules/admin/verbs/SDQL2/useful_procs.dm]: That would be **arc**cos then, wouldn't it?

Comment (Citinited) [reply code/modules/admin/verbs/SDQL2/useful_procs.dm]: no that's fine... there's a c between the a and r

Comment (Citinited) [reply code/modules/admin/verbs/SDQL2/useful_procs.dm]: Yeah, rogue c

Comment (Tayyyyyyy) [reply code/modules/admin/verbs/SDQL2/useful_procs.dm]: Oh, wow, I read that so many times and my brain just ignored it. Good catch!

**Comment by Fox-McCloud** [comment]
@Tayyyyyyy could you please fix the conflicts on this and also update anything to account for any new SS's/changes we added, please?

**Comment by Tayyyyyyy** [comment]
I've added tgstation's recently added dynamic subsystem parsing so no more need to update the PR for new SSes!

**Comment by Tayyyyyyy** [comment]
Looks like a problem on my end. Gonna go over and compare with the tg code and see what went wrong.

**Comment by AffectedArc07** [comment]
`tg restricts sdql2 queries from being executed via admin proc call, but we don't have this system`

We have this now

File: code/modules/admin/verbs/SDQL2/README.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+DELETE /mob/living/carbon/superior_animal/giant_spider WHERE loc.loc == marked
+```
+
+Here I used VV to mark the area they were in, and since `loc.loc == area`, voila.
+Only the spiders in a specific area are gone.
+
+Or you know if you want to catch spiders that crawled into lockers too (how even?)
+
+```sql
+DELETE /mob/living/carbon/superior_animal/giant_spider WHERE global.get_area(src) == marked
```
Comment (AffectedArc07) [inline code/modules/admin/verbs/SDQL2/README.md:88]: May want to make these the actual types we have here, `/mob/living/simple_animal/hostile/poison/giant_spider`

**Comment by variableundefined** [comment]
I have not heard from the author for a while, and the PR's conflicted, so I am closing until further notice.

This PR's approval still stands for any future iterations.

**Pull request closed without merging**